action.yml

```diff
@@ -34,64 +34,116 @@ inputs:
+      id: generate-token
       with:
-        app_id: ${{ inputs.github-app-id }}
-        private_key: ${{ inputs.github-app-key }}
+        app-id: ${{ inputs.github-app-id }}
+        private-key: ${{ inputs.github-app-key }}
+
+    - name: 🎭 Mock Convert token
+      if: ${{ env.ACT }}
+      id: mock-generate-token
+      shell: bash
+      run: |
+        echo "token=${{ env.TEST_TOKEN }}" >> $GITHUB_OUTPUT
 
     - name: ⬇️ Set up code
       uses: actions/checkout@v4
       with:
         show-progress: false
-        fetch-depth: 0
-        token: ${{ steps.generate_token.outputs.token }}
+        token: ${{ steps.generate-token.outputs.token || steps.mock-generate-token.outputs.token }}
+
+    - name: 🤔 Is this a release PR merge
+      id: is-release-pr
+      shell: bash
+      run: |
+        latest_commit_message=$(git log -1 --pretty=format:%s)
+        if [[ $latest_commit_message == "chore: release v"* ]]; then
+          echo "this workflow was triggered by a release PR"
+          echo "result=yes" >> $GITHUB_OUTPUT
+        else
+          echo "this workflow was not triggered by a release PR"
+          echo "result=no" >> $GITHUB_OUTPUT
+        fi
 
-    - name: ⏩ Get Next version
+    - name: ⏩ Get next version
+      if: ${{ !env.ACT }}
       id: get-next-version
       uses: agrc/get-next-version-action@v1
       with:
         prerelease: ${{ inputs.prerelease }}
         repo-token: ${{ inputs.repo-token }}
 
-    - name: 🚀 Create release
-      uses: google-github-actions/release-please-action@v4
-      id: release-please
+    - name: 🎭 Mock Get next version
+      if: ${{ env.ACT }}
+      id: mock-get-next-version
+      shell: bash
+      run: |
+        echo "version=9.9.9" >> $GITHUB_OUTPUT
+
+    - name: 📦 Setup Node.js
+      uses: actions/setup-node@v4
       with:
-        token: ${{ steps.generate_token.outputs.token }}
-        release-type: ${{ inputs.release-type }}
-        pull-request-title-pattern: 'chore: release v${version}'
-        changelog-types: >
-          [{"type":"feat","section":"🚀 Features","hidden":false},
-          {"type":"fix","section":"🐛 Bug Fixes","hidden":false},
-          {"type":"deps","section":"🌲 Dependencies","hidden":false},
-          {"type":"docs","section":"📖 Documentation Improvements","hidden":false},
-          {"type":"style","section":"🎨 Design Improvements","hidden":false},
-          {"type":"ci","section":"⚙️ Continuous Integration Changes","hidden":true},
-          {"type":"chore","section":"🧰 Maintenance","hidden":true}]
-        prerelease: ${{ inputs.prerelease }}
-        default-branch: ${{ github.ref_name }}
-        release-as: ${{ steps.get-next-version.outputs.version }}
-        extra-files: ${{ inputs.extra-files }}
+        node-version: lts/*
+
+    - name: Set up Release Please
+      shell: bash
+      run: |
+        git clone https://github.com/stdavis/release-please
+        cd release-please
+        git switch fix/merged-pr-searching
```

Comment:
**question**: is this the same branch in your pr to the upstream? Are you allowing collaborators to edit that branch? If so, are we ok with that?

Reply:
Great point. I switched to a separate branch that's not involved with the PR.